projects/hr-portal/src/app/services/data.service.ts

```diff
@@ -0,0 +1,16 @@
+const DATA_URL = "https://www.infragistics.com/grid-examples-data/data/hr/hr.json";
+
+export class DataService {
+    public async fetchData(): Promise<any> {
+      try {
+        const response = await fetch(DATA_URL);
+        const jsonData = await response.json();
+        return jsonData;
+      } catch (error) {
+        console.error('Error fetching HR data:', error);
+        throw error;
+      }
+    }
+  }
+
+  export {};
```

Comment:
Why exporting empty object?

Reply:
it was not accessible otherwise. This has now been changed to a singleton so it is no longer.